omnibus/config/software/libelf.rb

```diff
@@ -32,15 +32,13 @@
        if (!checks_passed)
 EOF
 )
-  env = {
-    "CFLAGS" => "-I#{install_dir}/embedded/include -O2 -pipe",
```

Comment:
I'm surprised not to see `-O3`, is it on purpose? Does `with_stardard_compiler_flags` change this?

Reply:
`-O3` sometimes has a reputation of using dangerous optimizations, so maybe that's the original reason.\

`with_standard_compiler_flags` also defaults to `-O2` though: https://github.com/DataDog/omnibus-ruby/blob/datadog-5.5.0/lib/omnibus/software.rb#L788 although it would be easier to trigger `-O3` for all configure based packages with this PR